### ISSUE TITLE
UI and UX is confusing

### ISSUE THREAD

**Issue by Nrgte** [AbdullahAlfaraj/Auto-Photoshop-StableDiffusion-Plugin#294]
So I've tried inpainting with the photoshop plugin on a couple of occations, but I always end up pretty frustrated. Even after going through the tutorials it's very confusing to me. Maybe that's just me, but I wanted to try and provide some feedback in case you're interested:

1. Firstly I feel like the plugin creates too many layers automatically, which makes me feel like I lose control and it's hard to understand which layer is for what. For example: The layer used for inpainting is automatically created and I have to use that layer. I would prefer it if I could create the layer myself, name it properly and then drag and drop it to the mask field.

2. Maybe I'm doing something wrong here, but when I delete some images through the viewer, the layer mask also gets deleted and I have to paint it again. 

3. If I delete a generated image layer directly, there doesn't seem to be a method to also delete that image in the viewer. It's stuck there forever or until I restart photoshop.

4. Generally it feels like I'm supposed to use the viewer instead of managing my layers, but if that's the case I find that very unintuitive for a photoshop workflow.

**Comment by albamuth** [comment]
I disagree with #1 - however I would have liked some instruction that I was supposed to use that auto-generated inpainting mask, it took me a while before I realized I needed to do my masking there.

**Comment by Woisek** [comment]
I while ago I suggested to look at stable.art on how they do it, because in stable.art masking needs only the lasso tool and hit generate. Everything else needed for the process later is done automatically. Extremely convenient. If Auto Photoshop StableDiffusion plugin would adopt this, it would be a perfect plugin for me.
But unfortunately there seems no interest in making this easier to use because I never got a feedback on that. 😔

**Comment by Nrgte** [comment]
> I disagree with #1 - however I would have liked some instruction that I was supposed to use that auto-generated inpainting mask, it took me a while before I realized I needed to do my masking there.

I mainly don't want the mask to get automatically deleted when I discard an image.
